plugins/outputs/azureblobstorage/helpermethods.go

```diff
@@ -0,0 +1,503 @@
+	_, er := blobClient.GetServiceProperties()
+	if er != nil {
+		log.Println("Error while validating blob client by calling GetServiceProperties()" + er.Error())
+		return er
+	}
+	return nil
+}
+
+func getJsonBlock(jsonObj BlockObject) (string, error) {
+	jsonBlock, er := json.Marshal(&jsonObj)
+	if er != nil {
+		return "", er
+	}
+	return string(jsonBlock), nil
+}
+
+//if event version = 2 then result is ver2v0
+func getEventVersionStr(eventVersion string) (string, error) {
+	eventVerInt, er := strconv.Atoi(eventVersion)
+	if er != nil {
+		log.Println("error while parsing event version." + eventVersion + " Event version should be in the format eg 2")
+		return "", er
+	}
+	eventVerStr := "ver" + strconv.Itoa(eventVerInt) + "v0"
+	return eventVerStr, nil
+}
+
+//takes in number of seconds and converts them into y=2015/m=05/d=03/h=00/m=00 format
+func getBaseTimeStr(baseTime string) (string, error) {
+	baseTimeInt, er := strconv.Atoi(baseTime) //TODO: use ParseInt
+	if er != nil {
+		log.Println("Error while Parsing baseTime" + baseTime)
+		return "", er
+	}
+	date := time.Unix(int64(baseTimeInt), 0)
+	baseTimeStr := "y=" + strconv.Itoa(date.Year()) + "/m=" + strconv.Itoa(int(date.Month())) + "/d=" + strconv.Itoa(date.Day()) + "/h=" + strconv.Itoa(date.Hour()) + "/m=" + strconv.Itoa(date.Minute())
+	return baseTimeStr, nil
+}
+
+func getBaseTimeMultipleOfInterval(baseTime string, interval string) (string, error) {
+
+	baseTimeInt64, er := strconv.ParseInt(baseTime, 10, 64)
+	if er != nil {
+		log.Println("Error while parsing baseTime " + baseTime + er.Error())
+		return "", er
+	}
+
+	intervalInt64, err := strconv.ParseInt(strings.Trim(interval, "s"), 10, 64)
+	if err != nil {
+		log.Println(" Error while parsing period." + interval)
+		log.Print(er.Error())
+		return "", er
+	}
+
+	//get baseTime in multiples of interval
+	baseTimeInt64 = baseTimeInt64 - (baseTimeInt64 % intervalInt64)
+	return strconv.FormatInt(baseTimeInt64, 10), nil
+}
+
+//blob path is a combination of resource id, identity hash and the base time
+//eg: Blob path: resourceId=<test_resource_id>/i=<agentIdentityHash>/y=2015/m=05/d=03/h=00/m=00/name=PT1H.json
+func getBlobPath(resourceId string, identityHash string, baseTime string, intervalISO8601 string) (string, error) {
+	baseTimeStr, er := getBaseTimeStr(baseTime)
+	if er != nil {
+		log.Println("Error while Parsing baseTime" + baseTime)
+		return "", er
+	}
+	blobPath := "resourceId=" + resourceId + "/i=" + identityHash + "/" + baseTimeStr + "/" + intervalISO8601 + ".json"
+	return blobPath, nil
+}
+func (azureBlobStorage *AzureBlobStorage) checkOldBlocksForBlob() (BlobPathBlockIds, error) {
+
+	var blockIdsBlobPath BlobPathBlockIds
+	raw, err := ioutil.ReadFile(azureBlobStorage.FileWithBlockIds)
+	if err != nil {
+		log.Println("!E Error while reading file " + azureBlobStorage.FileWithBlockIds + err.Error())
+		return blockIdsBlobPath, err
+	}
+
+	json.Unmarshal(raw, &blockIdsBlobPath)
```

Comment:
modify the statement to 
er:=json.Unmarshal(raw, &blockIdsBlobPath)
and add error check, and return error back to the calling method.
If an error occurs while reading the blockids which were written to the blob before telegraf crashed, then the calling method does't raise error instead ignores those block ids. 

Impact would be: these block ids will be lost from the blob, as every time a block is committed to a blob, the PutBlockList() API expects a list of all the block Ids which are to be present in this block, if any of the committed blockid is not present in the list then it's corresponding block will be lost.